### PR TITLE
Fix class too large error with string constants class due to constant pool size exceeds the limit

### LINKED ISSUE #38796: [Bug]: Class too large exception in a large Ballerina project while bal build
### Description

While building a Ballerina Project with 250+ Ballerina files and each .bal file having at least 2 const defined with in, throws the following exception,

`[2022-11-23 15:37:08,733] SEVERE {b7a.log.crash} - Class too large: wso2healthcare/healthcare$0046fhir$0046r4/0/constants/$_string_constants 
org.objectweb.asm.ClassTooLargeException: Class too large: wso2healthcare/healthcare$0046fhir$0046r4/0/constants/$_string_constants
        at org.objectweb.asm.ClassWriter.toByteArray(ClassWriter.java:538)
        at org.wso2.ballerinalang.compiler.bir.codegen.split.constants.JvmBStringConstantsGen.generateConstantInit(JvmBStringConstantsGen.java:130)
        at org.wso2.ballerinalang.compiler.bir.codegen.split.JvmConstantsGen.generateConstants(JvmConstantsGen.java:91)
        at org.wso2.ballerinalang.compiler.bir.codegen.JvmPackageGen.generate(JvmPackageGen.java:809)
        at org.wso2.ballerinalang.compiler.bir.codegen.CodeGenerator.generate(CodeGenerator.java:96)
        at org.wso2.ballerinalang.compiler.bir.codegen.CodeGenerator.generate(CodeGenerator.java:71)
        at io.ballerina.projects.JBallerinaBackend.performCodeGen(JBallerinaBackend.java:309)
        at io.ballerina.projects.ModuleContext.generateCodeInternal(ModuleContext.java:473)
        at io.ballerina.projects.ModuleCompilationState$4.generatePlatformSpecificCode(ModuleCompilationState.java:132)
        at io.ballerina.projects.ModuleContext.generatePlatformSpecificCode(ModuleContext.java:381)
        at io.ballerina.projects.JBallerinaBackend.performCodeGen(JBallerinaBackend.java:165)
        at io.ballerina.projects.JBallerinaBackend.<init>(JBallerinaBackend.java:134)
        at io.ballerina.projects.JBallerinaBackend.lambda$from$0(JBallerinaBackend.java:112)
        at java.base/java.util.HashMap.computeIfAbsent(HashMap.java:1133)
        at io.ballerina.projects.PackageCompilation.getCompilerBackend(PackageCompilation.java:169)
        at io.ballerina.projects.JBallerinaBackend.from(JBallerinaBackend.java:111)
        at io.ballerina.cli.task.CompileTask.execute(CompileTask.java:130)
        at io.ballerina.cli.TaskExecutor.executeTasks(TaskExecutor.java:40)
        at io.ballerina.cli.cmd.BuildCommand.execute(BuildCommand.java:252)
        at java.base/java.util.Optional.ifPresent(Optional.java:183)
        at io.ballerina.cli.launcher.Main.main(Main.java:51)`

However, when `.bal` files reduced around to 50, the `bal build` completed successfully.

### Steps to Reproduce

The issue raised when building a large Ballerina project with more than 250 `.bal` files where each file has at least 2 `const` defined within.

### Affected Version(s)

Tested on Ballerina distributions: 2201.1.2, 2201.2.1, 2201.2.3

### OS, DB, other environment details and versions

OS: Ubuntu 20.04

### Related area

-> Compilation

### Related issue(s) (optional)

https://stackoverflow.com/questions/74546037/class-too-large-exception-in-a-large-ballerina-project-while-bal-build

### Suggested label(s) (optional)

_No response_

### Suggested assignee(s) (optional)

_No response_

### PR DESCRIPTION
**Pull request by gabilang**
When a module has large number of string constants (more than 13000), compilation gives class too large error due to the constant pool of generated string constant class exceeds its limit. This problem has been solved by splitting the string constant class into multiple classes when the string constants count gets larger (String constants count of a class has been limited to 5000).

Fixes #38796 

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: tests/jballerina-unit-test/src/test/resources/test-src/jvm/tooLargeStringConstantClass/Ballerina.toml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+[build-options]
+observabilityIncluded = false
```
Comment (Nadeeshan96) [inline tests/jballerina-unit-test/src/test/resources/test-src/jvm/tooLargeStringConstantClass/Ballerina.toml:2]: Missing new line

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmCodeGenUtil.java [hunk 2]
```diff
@@ -769,6 +773,11 @@ public static void loadConstantValue(BType bType, Object constVal, MethodVisitor
         }
     }
 
+    private static String getStringConstantsClass(int varIndex, JvmConstantsGen jvmConstantsGen) {
+        int classIndex = (varIndex / MAX_STRINGS_PER_METHOD);
```
Comment (warunalakshitha) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmCodeGenUtil.java:777]: We do not need parentheses here?

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/split/constants/JvmBStringConstantsGen.java [hunk 3]
```diff
@@ -246,35 +235,54 @@ private void visitBStringField(ClassWriter cw, String varName) {
...
+        String constantClassName = null;
         int bStringCount = 0;
-        int methodCount = 0;
-        for (Map.Entry<String, String> entry : bStringVarMap.entrySet()) {
+        for (Map.Entry<String, Integer> entry : bStringVarIndexMap.entrySet()) {
+            String bString = entry.getKey();
+            int varIndex = entry.getValue();
+            String bStringVarName = B_STRING_VAR_PREFIX + varIndex;
+
+            int classIndex = (varIndex / MAX_STRINGS_PER_METHOD);
```
Comment (warunalakshitha) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/split/constants/JvmBStringConstantsGen.java:249]: We can remove Parentheses here

**Review by warunalakshitha — APPROVED** [review]
LGTM

**Pull request merged**